File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/plugin/evaluate/EntityValidationPluginEvaluationOptions.js
define(["stjs", "as/dto/plugin/evaluate/PluginEvaluationOptions"], function (
  stjs,
  PluginEvaluationOptions
) {
  var EntityValidationPluginEvaluationOptions = function () {};
  stjs.extend(
    EntityValidationPluginEvaluationOptions,
    PluginEvaluationOptions,
    [PluginEvaluationOptions],
    function (constructor, prototype) {
      prototype["@type"] =
        "as.dto.plugin.evaluate.EntityValidationPluginEvaluationOptions";
      constructor.serialVersionUID = 1;
      prototype.objectId = null;
      prototype.isNew = null;

      prototype.getObjectId = function () {
        return this.objectId;
      };
      prototype.setObjectId = function (objectId) {
        this.objectId = objectId;
      };
      prototype.isNew = function () {
        return this.isNew;
      };
      prototype.setNew = function (isNew) {
        this.isNew = isNew;
      };
      prototype.getIsNew = function () {
        return this.isNew;
      };
      prototype.setIsNew = function (isNew) {
        this.isNew = isNew;
      };
    },
    {
      objectId: "IObjectId",
    }
  );
  return EntityValidationPluginEvaluationOptions;
});

File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/plugin/evaluate/EntityValidationPluginEvaluationOptions.java
package ch.ethz.sis.openbis.generic.asapi.v3.dto.plugin.evaluate;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.ObjectToString;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.id.IObjectId;
import ch.systemsx.cisd.base.annotation.JsonObject;

@JsonObject("as.dto.plugin.evaluate.EntityValidationPluginEvaluationOptions")
public class EntityValidationPluginEvaluationOptions extends PluginEvaluationOptions
{

    private static final long serialVersionUID = 1L;

    @JsonProperty
    private IObjectId objectId;

    @JsonProperty
    private boolean isNew;

    @JsonIgnore
    public IObjectId getObjectId()
    {
        return objectId;
    }

    public void setObjectId(IObjectId objectId)
    {
        this.objectId = objectId;
    }

    @JsonIgnore
    public boolean isNew()
    {
        return isNew;
    }

    public void setNew(boolean isNew)
    {
        this.isNew = isNew;
    }

    @Override
    protected ObjectToString toObjectToString()
    {
        return super.toObjectToString().append("objectId", objectId).append("isNew", isNew);
    }

    @Override
    public String toString()
    {
        return toObjectToString().toString();
    }

}
